File: boat-sample/src/test/java/sample/java/xyz/srclab/common/codec/CodecSample.java
package sample.java.xyz.srclab.common.codec;

import org.testng.annotations.Test;
import xyz.srclab.common.codec.Codec;
import xyz.srclab.common.codec.aes.AesKeys;
import xyz.srclab.common.test.TestLogger;

import javax.crypto.SecretKey;

public class CodecSample {

    private static final TestLogger logger = TestLogger.DEFAULT;

    @Test
    public void testCodec() {
        String password = "hei, xiongdi, womenhaojiubujiannizainali";
        String messageBase64 = "aGVpLCBwZW5neW91LCBydWd1b3poZW5kZXNoaW5pcWluZ2Rhemhhb2h1";
        SecretKey secretKey = AesKeys.newKey(password);

        //Use static
        String message = Codec.decodeBase64String(messageBase64);
        byte[] encrypt = Codec.aesCipher().encrypt(secretKey, message);
        String decrypt = Codec.aesCipher().decryptToString(secretKey, encrypt);
        //hei, pengyou, ruguozhendeshiniqingdazhaohu
        logger.log("decrypt: {}", decrypt);

        //Use chain
        encrypt = Codec.forData(messageBase64).decodeBase64().encryptAes(secretKey).doFinal();
        decrypt = Codec.forData(encrypt).decryptAes(secretKey).doFinalToString();
        //hei, pengyou, ruguozhendeshiniqingdazhaohu
        logger.log("decrypt: {}", decrypt);
    }
}

File: boat-sample/src/test/kotlin/sample/kotlin/xyz/srclab/common/codec/CodecSample.kt
package sample.kotlin.xyz.srclab.common.codec

import org.testng.annotations.Test
import xyz.srclab.common.codec.Base64Codec
import xyz.srclab.common.codec.CipherCodec
import xyz.srclab.common.codec.Codecing.Companion.startCodec
import xyz.srclab.common.codec.aes.toAesKey
import xyz.srclab.common.test.TestLogger

class CodecSample {

    @Test
    fun testCodec() {
        val password = "hei, xiongdi, womenhaojiubujiannizainali"
        val messageBase64 = "aGVpLCBwZW5neW91LCBydWd1b3poZW5kZXNoaW5pcWluZ2Rhemhhb2h1"
        val secretKey = password.toAesKey()

        //Use static
        val message: String = Base64Codec.decodeToString(messageBase64)
        var encrypt = CipherCodec.aes().encrypt(secretKey, message)
        var decrypt = CipherCodec.aes().decryptToString(secretKey, encrypt)
        //hei, pengyou, ruguozhendeshiniqingdazhaohu
        logger.log("decrypt: {}", decrypt)

        //Use chain
        encrypt = messageBase64.startCodec().decodeBase64().encryptAes(secretKey).doFinal()
        decrypt = encrypt.startCodec().decryptAes(secretKey).doFinalString()
        //hei, pengyou, ruguozhendeshiniqingdazhaohu
        logger.log("decrypt: {}", decrypt)
    }

    companion object {
        private val logger = TestLogger.DEFAULT
    }
}